tests/integration/test_inspect_table.py

```diff
@@ -938,3 +947,37 @@ def test_inspect_all_manifests(spark: SparkSession, session_catalog: Catalog, fo
     lhs = spark.table(f"{identifier}.all_manifests").toPandas()
     rhs = df.to_pandas()
     assert_frame_equal(lhs, rhs, check_dtype=False)
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize("format_version", [1, 2])
+def test_inspect_all_files(
+    spark: SparkSession, session_catalog: Catalog, arrow_table_with_null: pa.Table, format_version: int
+) -> None:
+    identifier = "default.table_metadata_files"
+
+    tbl = _create_table(session_catalog, identifier, properties={"format-version": format_version})
```

Comment:
Instead of setting the properties below, we can also do this during table creation:
```suggestion
    tbl = _create_table(session_catalog, identifier, properties={"format-version": format_version, "write.delete.mode": "merge-on-read", "write.update.mode": "merge-on-read"})
```

This avoids another round trip, speeding up the tests :)

Reply:
If I do this, then I get below error for format version 1 `pyspark.errors.exceptions.captured.IllegalArgumentException: Deletes are supported in V2 and above`.
Hence I'm setting the properties for format version 2 only.